lib/reek/context/code_context.rb

```diff
@@ -16,13 +15,16 @@ module Context
     #
     # :reek:TooManyMethods: { max_methods: 19 }
     # :reek:TooManyInstanceVariables: { max_instance_variables: 8 }
+    # :reek:Attribute
     class CodeContext
       extend Forwardable
       delegate each_node: :exp
       delegate %i(name type) => :exp
-      delegate %i(visibility visibility= non_public_visibility?) => :visibility_tracker
 
-      attr_reader :children, :context, :exp, :statement_counter, :visibility_tracker
+      attr_accessor :visibility
+      attr_reader :children, :context, :exp
+      attr_reader :statement_counter
```

Comment:
It seemed nicer because statement_counter is a tool while the other two are properties. I'm not super attached to this change. Ideally, I would not even expose statement_counter, but that is for another PR.